libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -967,12 +983,26 @@ void AP_GPS::update_primary(void)
             _last_instance_swap_ms = now;
             continue;
         }
+        if (state[i].status < state[primary_instance].status) {
```

Comment:
maybe treat status==3 as status==4 when we prefer one GPS?

Reply:
removed